### DISCUSSION TITLE
Optimization (?) of startup

### DISCUSSION THREAD

**Discussion by yonib** [kartoza/docker-postgis#466] [Ideas]
In first wakeup from "docker-compose up -d", the system arrives at setup-database.sh and approx. in line 77 in the script it calls the function "non_root_permissions postgres postgres".
This function looks like this:
```
function non_root_permission() {
  USER="$1"
  GROUP="$2"
  path_envs=("${DATADIR}" "${WAL_ARCHIVE}" "${SCRIPTS_LOCKFILE_DIR}" "${CONF_LOCKFILE_DIR}" "${EXTRA_CONF_DIR}" "${SSL_DIR}" "${POSTGRES_INITDB_WALDIR}")
  for dir_names in "${path_envs[@]}";do
    if [ ! -z "${dir_names}" ];then
      directory_checker "${dir_names}"
    fi
  done
  services=("/usr/lib/postgresql/" "/etc/" "/var/log/postgresql" "/var/run/!(secrets)" "/var/lib/" "/usr/bin" "/tmp" "/scripts")
  for paths in "${services[@]}"; do
    directory_checker "${paths}"
  done
  chmod -R 750 "${DATADIR}" ${WAL_ARCHIVE}

}

```
basically, it calls directory_checker function which checks if the directory has the postgres:postgres owner/group pair, and it not - it uses chown with -R flag to set the ownerships of the directory recoursively.

it does so for small directories like "/scripts", but also for very large directories like "/usr/lib/postgresql/ ", "/var/lib/" and "/usr/bin/"...

my question is why don't chown these directories upon image creation ? this will leave directory_checker with much less work when waking up from "docker-compose up -d"

Thanks !

Yoni

**Comment by NyakudyaA** [comment]
What would happen if you manage to change the image from root to postgres to another user. If you constantly using the same user the check is very fast, it will only slow down when the check is going to chown
